Add documentation for the type::range function (#440)
surrealdb/docs.surrealdb.com: doc-surrealdb_versioned_docs/version-1.x/surrealql/functions/type.mdx

LET $id = "tobie";
RETURN type::thing($tb, $id);
```
<br />

## `type::range`

The `type::range` function converts a value into a record range.

```surql title="API DEFINITION"
type::range(any, any, any, object) -> thing
```

> This function takes on up two four argument. 
> The first is the table of range. 
> The second and third are the beginning and end of the range. If either is NULL or not specified, the range will be unbounded in the direction that is not specified.
> The fourth argument is an object which can be used to define how the range is bounded. 
> If an object is specified as the fourth argument with the property `begin` or `end` containing the value `"included"` or `"excluded"` the ranges boundd will be updated accordingly.
> If no bounds are specified the function will default to including the begin bound and excluding the end bound.

Reviewer: ```suggestion
This function takes up to four arguments.

The first is the table name for the range. The second and third are the beginning and end of the range. If either is NULL or not specified, the range will be unbounded in the direction that is not specified.

The fourth is an object which can be used to define how the range is bounded. If it contains the properties `begin` or `end` (containing the value `"included"` or `"excluded"`) the ranges bound will be updated accordingly.

If no bounds are specified, the function will default to including the begin bound and excluding the end bound.
```

The following example shows this function, and its output, when used in a [`RETURN`](/docs/surrealdb/surrealql/statements/return) statement:

```surql
LET $tb = "product_price";
LET $begin = "10";
LET $end = "100";
// create the range `product_price:10>..=100`
RETURN type::thing($tb, $begin, $end, { begin: "excluded", end: "included" });

> ```

Reviewer: Here I've found that a second code block with bash title="Response" after the three ticks looks nice for showing the output. Could you add that and paste the actual output into there?

LET $id = "tobie";
RETURN type::thing($tb, $id);
```
<br />

## `type::range`

The `type::range` function converts a value into a record range.

```surql title="API DEFINITION"
type::range(any, any, any, object) -> thing
```
This function takes up to four arguments.

The first is the table name for the range. The second and third are the beginning and end of the range. If either is NULL or not specified, the range will be unbounded in the direction that is not specified.

The fourth is an object which can be used to define how the range is bounded. If it contains the properties `begin` or `end` (containing the value `"included"` or `"excluded"`) the ranges bound will be updated accordingly.

If no bounds are specified, the function will default to including the begin bound and excluding the end bound.

The following example shows this function, and its output, when used in a [`RETURN`](/docs/surrealdb/surrealql/statements/return) statement:

```surql
LET $tb = "product_price";
LET $begin = "10";
LET $end = "100";
// create the range `product_price:10>..=100`

> RETURN type::thing($tb, $begin, $end, { begin: "excluded", end: "included" });

Reviewer 2: ```suggestion
RETURN type::range($tb, $begin, $end, { begin: "excluded", end: "included" });
```
